text/0035-physics.md

```diff
@@ -0,0 +1,175 @@
+
+## Why we do not synchronize velocity
+
+There are two reasons why we would need to synchronize velocity.
+1. Prediction: velocity allows for a more accurate prediction.
+2. Extrapolation in case of packet loss.
+
+We do not support prediction currently so 1. is not relevant. Regarding extrapolation while this might improve the smoothness of the Rigidbody when extrapolating in case of a lag due to packet loss it is non trivial to implement this without resulting in jitter. What we would need to do here is transition the Rigidbody from a simply replicating mode into a dynamic Rigidbody again and run the physics simulation to extrapolate. We currently do not have a good way to make this transition smooth because we are not in control of the FixedUpdate and the physics simulation. Instead simply using the NetworkTransforms extrapolation mode results in a more smooth experience and keeps our implementation simple.
+
+## Why we do not synchronize triggers and collision events
+
+1. Collision and Trigger events contain information in reference types (classes) which is not something we could properly serialize. In addition these events contain information about the components and GameObject involved in the collision. We currently have no way to serialize that information over the network.
+
+2. Assuming we could serialize the data of the collision event there's an other issue. When sending collisions we could either send them reliable or unreliable both of them have downsides:
+    1. Unreliable: Would kind of defeat the purpose because collisions and triggers would still not be fully reliable
+    2. Reliable: If the packet gets dropped the collision even will be delayed compared to the position sync.
+
+Because of the two reasons above we will not do collision and trigger synchronization above and instead ask people to write their own using reliable RPCs if they need a collision event to be reliably synced to all clients. 
+
+## Switching Kinematic Mode at Runtime
+
+Sometimes for gameplay reasons switching the kinematic mode of the Rigidbody is needed. We do support switching the kinematic mode of the Rigidbody on the controller of the Rigidbody. The kinematic mode of the Rigidbody will be synchronized with a NetworkVariable to all other peers. This allows switching the owner of the Rigidbody 
+
+## Synchronizing Rigidbody Parameters
+
+We will not synchronize any Rigidbody parameters besides `IsKinematic`. For instance we will not synchronize the Mass or the Drag or the Constraints of a Rigidbody. The reasons we do not synchronize them are to save bandwidth and to keep the use cases of Rigidbody flexible. If synchronization of a specific Rigidbody parameter is needed users can add this with a custom script using a NetworkVariable.
```

Comment:
"to save bandwidth" --> maybe I'm missing something, that's a few bytes at spawn, then nothing if it doesn't change, no?
Not sure I understand the flexibility bit.

Reply:
People might change these values but not change velocities. For instance it is quite common to adjust the mass or add constraints to a RigidBody at runtime. If the `controller` of the body never changes there is no need to synchronize those values.

Reply:
that's a dirty check though no? That's different from saying we'll **never** sync these values. It's a bit like what Fatih did with NetworkTransform's state and per-axis synchronisation.